share/gpodder/extensions/tagging.py

```diff
@@ -224,22 +262,60 @@ def __init__(self, container):
         EasyID3.RegisterTextKey("subtitle", "TIT3")
         EasyMP4Tags.RegisterTextKey("comments", "desc")
         EasyMP4Tags.RegisterFreeformKey("subtitle", "SUBTITLE")
+        self.art_filetypes = [
+            "JPEG",
+            "PNG",
+        ]
 
     def on_episode_downloaded(self, episode):
+        # Ensure we're within the bounds of the list
+        if self.container.config.episode_coverart_filetype > (len(self.art_filetypes) - 1):
+            self.container.config.episode_coverart_filetype = 0
+
+        config_filetype = self.art_filetypes[self.container.config.episode_coverart_filetype].upper()
+
         info = self.read_episode_info(episode)
         if info['filename'] is None:
             return
 
-        self.write_info2file(info, episode)
+        extracted_image = self.get_embeddedart(info, episode)
+        channel_image_filename = self.get_channelart(episode.channel)
+        try:
+            with open(channel_image_filename, 'rb') as f:
+                channel_image = f.read()
+        except:
+            logger.error("problems reading channel image!")
```

Comment:
this way we get the stacktrace:
```suggestion
            logger.exception("problems reading channel image!")
```

Reply:
or if you don't want the stacktrace
```suggestion
            logger.error("problems reading channel image: %r!", sys.exc_info()[1])
```